### ISSUE TITLE
[24.1] JFR integration tests fail for Mandrel for JDK 23 - JDK events initialization

### ISSUE THREAD

**Issue by jerboaa** [Karm/mandrel-integration-tests#268]
Latest Mandrel integration tests fail the JFR integration test with this warning:

```
[warn][jfr,system] Could not initialize JDK events. jfrTracing
```

... which usually indicates a larger problem, so we should investigate.

Example test output:

```
Error:    JFRTest.jfrOptionsSmokeTest:784->jfrOptionsSmoke:922 build-and-run.log log should not contain error or warning lines that are not whitelisted. See /home/runner/work/mandrel/mandrel/mandrel-integration-tests/testsuite/target/archived-logs/org.graalvm.tests.integration.JFRTest/jfrOptionsSmokeTest/build-and-run.log and check these offending lines: 
[warn][jfr,system] Could not initialize JDK events. jfrTracing ==> expected: <true> but was: <false>
Error:    JFRTest.jfrPerfTest:185->jfrPerfTestRun:234 build-and-run.log log should not contain error or warning lines that are not whitelisted. See /home/runner/work/mandrel/mandrel/mandrel-integration-tests/testsuite/target/archived-logs/org.graalvm.tests.integration.JFRTest/jfrPerfTest/build-and-run.log and check these offending lines: 
[warn][jfr,system] Could not initialize JDK events. jfrTracing
```

See this CI link:
https://github.com/graalvm/mandrel/actions/runs/9720067442/job/26831418361#step:11:20582

**Comment by jerboaa** [comment]
@roberttoyonaga Could you please take a look at this? Thanks!

**Comment by jerboaa** [comment]
Seems related to https://bugs.openjdk.org/browse/JDK-8331876 new in `JDK 23+26`.

**Comment by jerboaa** [comment]
It doesn't seem to be happening for a Mandrel build with `JDK 24+3`. Looks like we'd need a backport of https://github.com/oracle/graal/pull/9212

**Comment by jerboaa** [comment]
Steps to reproduce:

```
export JAVA_HOME=/path/to/mandrel-build-based-on-jdk23
export GRAALVM_HOME=$JAVA_HOME
export PATH=$JAVA_HOME/bin:$PATH
export FAIL_ON_PERF_REGRESSION=false
mvn clean verify -Dquarkus.version=999-SNAPSHOT -Ptestsuite -Dtest=JFRTest#jfrPerfTest
```

**Comment by roberttoyonaga** [comment]
> @roberttoyonaga Could you please take a look at this? Thanks!

Yup I can look at this!

**Comment by jerboaa** [comment]
No longer an issue. Fixed with  graalvm/mandrel#762.
